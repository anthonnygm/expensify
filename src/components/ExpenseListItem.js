import React from 'react';
import {Link} from 'react-router-dom';
import * as moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({id, description, note, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`} />
        <h3>{description}</h3>
        <p>
            {numeral(amount / 100).format('R$0.0,00')}-
            {moment(createdAt).format('MMMM Do, YYYY')}
        </p>
    </div>
);

/**
 * import {connect} from 'react-redux';
 * import {removeExpense} from '../actions/expenses';
 * , dispatch
 * export default connect()(ExpenseListItem);
 * <button onClick={() => { dispatch(removeExpense({id})); }} >
 * Remove
 * </button>
 */
export default ExpenseListItem;
