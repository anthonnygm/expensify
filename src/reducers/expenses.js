// Expenses Reducer
const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ]
            break;
        case 'REMOVE_EXPENSE':
            return state.filter(({id}) => id !== action.id);
            break;
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    };
                } else {
                    return expense;
                }
            });
            break;
        default:
            return state;
    }
};

