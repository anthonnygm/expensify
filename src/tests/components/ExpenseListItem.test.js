import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render ExpenseListItem correctly', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    console.log(wrapper);
    expect(wrapper).toMatchSnapshot();
});
