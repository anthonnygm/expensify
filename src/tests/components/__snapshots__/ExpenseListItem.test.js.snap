// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<div>
  <h3>
    Gum
  </h3>
  <p>
    195
     - 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
